.github/workflows/build-docker.yml

```diff
@@ -22,26 +23,38 @@ jobs:
         target: [ release, test ]
     runs-on: ubuntu-latest-8-cores
     permissions:
-      contents: write    
+      contents: write # to update the GH release w/ Docker tags
+      packages: write # to upload images to ghcr
+      id-token: write
     steps:
       - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac #v4
         with:
           fetch-depth: 0
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 #v3
-      - name: DockerHub Login
-        uses: docker/login-action@v3
+
+      # - name: DockerHub Registry Login
+      #   uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d
+      #   with:
+      #     username: ${{ secrets.DOCKER_HUB_USERNAME }}
+      #     password: ${{ secrets.DOCKER_HUB_TOKEN }}
```

Comment:
Can we entirely remove this? We shouldn't commit commented code unless not absolutely necessary. If so, please add a top comment explaining why this is commented out and when it should be used.